app/src/main/java/org/mozilla/fenix/ext/Fragment.kt

```diff
@@ -128,3 +133,14 @@ fun Fragment.removeSecure() {
         WindowManager.LayoutParams.FLAG_SECURE,
     )
 }
+
+/**
+ * Register a request to start an activity for result.
+ */
+fun Fragment.registerForActivityResult(onSuccess: (result: ActivityResult) -> Unit): ActivityResultLauncher<Intent> {
+    return registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+        if (result.resultCode == Activity.RESULT_OK) {
```

Comment:
It could actually cause bugs, in case the receiver expects and handles any other results but the success or is waiting for any result. Maybe adding a second callback for cancelled result for edge cases (like `BaseVoiceSearchActivity`)?

Reply:
Sure, I'll add the onFailure function and set it to empty by default.  Currently from testing, I don't see any need to handle results when failed.